File: 58. Length of Last Word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        using namespace std;
      //find end of string
        int ending = s.length() - 1;
      //find where last word ends, skip over spaces at end
        while (ending >= 0 && s[ending] == ' ')
            ending--;
        
        int start = ending;
      //find beginning of last word
        while (start >= 0 && s[start] != ' ')
            start--;
      
        return ending - start;
    }
};

File: 58. Length of Last Word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:

        ending = len(s)-1

        while ending >= 0 and s[ending] == ' ':
            ending = ending - 1
        
        start = ending

        while start >= 0 and s[start] != ' ':
            start = start - 1

        return (ending - start)
